slick.dataview.js

```diff
@@ -1265,67 +1267,168 @@
       _grid = grid;
       var inHandler;
       selectedRowIds = self.mapRowsToIds(grid.getSelectedRows());
-      var onSelectedRowIdsChanged = new Slick.Event();
 
+      /** @param {Array} rowIds */
       function setSelectedRowIds(rowIds) {
-        if (selectedRowIds.join(",") == rowIds.join(",")) {
-          return;
+        if (rowIds === false) {
+          selectedRowIds = [];
+        } else {
+          if (selectedRowIds.sort().join(",") !== rowIds.sort().join(",")) {
+            selectedRowIds = rowIds;
+          }
         }
-
-        selectedRowIds = rowIds;
-
-        onSelectedRowIdsChanged.notify({
-          "grid": grid,
-          "ids": selectedRowIds,
-          "dataView": self
-        }, new Slick.EventData(), self);
       }
 
       function update() {
-        if (selectedRowIds.length > 0) {
+        if (selectedRowIds.length > 0 && !inHandler) {
           inHandler = true;
           var selectedRows = self.mapIdsToRows(selectedRowIds);
           if (!preserveHidden) {
-            setSelectedRowIds(self.mapRowsToIds(selectedRows));
+            var selectedRowsChangedArgs = {
+              "grid": _grid,
+              "ids": self.mapRowsToIds(selectedRows),
+              "dataView": self
+            };
+            selectedRowChangedFn(selectedRowsChangedArgs);
+            onSelectedRowIdsChanged.notify(Object.assign(selectedRowsChangedArgs, {
+              "filteredIds": self.getAllSelectedFilteredIds(),
```

Comment:
added back `filteredIds` that was originally added in first commit but later removed in PR #689, this propery is useful when using Pagination and we want to know all Ids selected